move to .net core!

who can move it to .net core?
It's very important for me.Anyone help?

I needed an update to be compatible with the new .net core too.  Building to Windows Store in Unity using this component for UnityMaterialUI, and ran into some errors that are keeping me from porting over, thus we can't run our app on Hololens like it needs to. Here are the problems:
Assets\MaterialUI\Utils\LitJson\JsonReader.cs(405,24): error CS1061: 'TextReader' does not contain a definition for 'Close' and no extension method 'Close' accepting a first argument of type 'TextReader' could be found
~~I fixed this one by replacing reader.Close (); with reader.Dispose();

Assets\MaterialUI\Utils\LitJson\JsonMapper.cs(173,22)&(205,22): error CS1061: 'Type' does not contain a definition for 'GetInterface' and no extension method 'GetInterface' accepting a first argument of type 'Type' could be found

...\LitJson\JsonMapper.cs(317,31): error CS1061: 'Type' does not contain a definition for 'IsClass' and no extension method 'IsClass' accepting a first argument of type 'Type' could be found
...\LitJson\JsonMapper.cs(372,32): error CS1061: 'Type' does not contain a definition for 'IsEnum' and no extension method 'IsEnum' accepting a first argument of type 'Type' could be found

I tried using the legacy versions of the System classes by doing this:
#if NETFX_CORE
using Type = WinRTLegacy.Type;
#else
using Type = System.Type;
#endif
But sadly it wasn't one of the available ones, and I'm not exactly sure what's going on in this code to be able to replace it with alternative methods.  Anybody else smart enough to update this for Microsoft's moot changes that make our lives more difficult?  Probably not too hard, but this isn't my realm.  Thanks..

--- a/.gitignore
+++ b/.gitignore
@@ -1,3 +1,10 @@
 /bin
 /src/LitJson/AssemblyInfo.cs
 /website
+[Oo]bj/
+[Bb]in/
+.vscode/
+[Tt]ools/*
+![Tt]ools/packages.config
+[Aa]rtifacts/
+.[Dd]otnet/

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,0 +1,24 @@
+language: csharp
+os:
+  - osx
+  - linux
+
+# Ubuntu 14.04
+sudo: required
+dist: trusty
+
+# OS X 10.12
+osx_image: xcode9.2
+
+mono:
+  - 4.4.2
+
+dotnet: 2.1.3
+
+before_install:
+  - git fetch --unshallow # Travis always does a shallow clone, but GitVersion needs the full history including branches and tags
+  - git config remote.origin.fetch "+refs/heads/*:refs/remotes/origin/*"
+  - git fetch origin  
+
+script:
+  - ./build.sh --target=Test

--- a/LitJSON.sln
+++ b/LitJSON.sln
@@ -1,0 +1,48 @@
+
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio 15
+VisualStudioVersion = 15.0.26124.0
+MinimumVisualStudioVersion = 15.0.26124.0
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "LitJSON", "src\LitJson\LitJSON.csproj", "{CFBC33D5-BB2D-4F22-B157-0862C4A77039}"
+EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "LitJSON.Tests", "test\LitJSON.Tests.csproj", "{E031C1B4-CD87-475E-9283-4571435C65DB}"
+EndProject
+Global
+	GlobalSection(SolutionConfigurationPlatforms) = preSolution
+		Debug|Any CPU = Debug|Any CPU
+		Debug|x64 = Debug|x64
+		Debug|x86 = Debug|x86
+		Release|Any CPU = Release|Any CPU
+		Release|x64 = Release|x64
+		Release|x86 = Release|x86
+	EndGlobalSection
+	GlobalSection(SolutionProperties) = preSolution
+		HideSolutionNode = FALSE
+	EndGlobalSection
+	GlobalSection(ProjectConfigurationPlatforms) = postSolution
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|x64.ActiveCfg = Debug|x64
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|x64.Build.0 = Debug|x64
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|x86.ActiveCfg = Debug|x86
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Debug|x86.Build.0 = Debug|x86
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|Any CPU.Build.0 = Release|Any CPU
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|x64.ActiveCfg = Release|x64
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|x64.Build.0 = Release|x64
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|x86.ActiveCfg = Release|x86
+		{CFBC33D5-BB2D-4F22-B157-0862C4A77039}.Release|x86.Build.0 = Release|x86
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|x64.ActiveCfg = Debug|x64
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|x64.Build.0 = Debug|x64
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|x86.ActiveCfg = Debug|x86
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Debug|x86.Build.0 = Debug|x86
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|Any CPU.Build.0 = Release|Any CPU
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|x64.ActiveCfg = Release|x64
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|x64.Build.0 = Release|x64
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|x86.ActiveCfg = Release|x86
+		{E031C1B4-CD87-475E-9283-4571435C65DB}.Release|x86.Build.0 = Release|x86
+	EndGlobalSection
+EndGlobal

--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,14 +1,9 @@
-version: 1.0.{build}
-image: Visual Studio 2017
-install:
-- ps: >-
-    choco install make
+# Operating system (build VM template)
+os: Visual Studio 2017
 
-
-    mkdir bin
+# Build script
 build_script:
-- cmd: >-
-    cd build/make
+  - ps: .\build.ps1  --target="AppVeyor" --verbosity=Verbose
 
-    make CSC='"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\Roslyn\csc.exe"'
+# Tests
 test: off

--- a/build.cake
+++ b/build.cake
@@ -1,0 +1,217 @@
+
+#tool "nuget:https://api.nuget.org/v3/index.json?package=GitVersion.CommandLine&version=3.6.2"
+#tool nuget:?package=NUnit.ConsoleRunner&version=3.4.0
+
+///////////////////////////////////////////////////////////////////////////////
+// ARGUMENTS
+///////////////////////////////////////////////////////////////////////////////
+
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+
+//////////////////////////////////////////////////////////////////////
+// PARAMETERS
+//////////////////////////////////////////////////////////////////////
+
+DotNetCoreMSBuildSettings msBuildSettings = null;
+string  version = null,
+        semVersion = null,
+        milestone = null;
+
+///////////////////////////////////////////////////////////////////////////////
+// SETUP / TEARDOWN
+///////////////////////////////////////////////////////////////////////////////
+
+Setup(ctx =>
+{
+    Information("Calculating Semantic Version");
+    if (!BuildSystem.IsLocalBuild)
+    {
+        GitVersion(new GitVersionSettings{
+            OutputType = GitVersionOutput.BuildServer
+        });
+    }
+
+    CopyFile("./src/LitJson/AssemblyInfo.cs.in", "./src/LitJson/AssemblyInfo.cs");
+
+    GitVersion assertedVersions = GitVersion(new GitVersionSettings
+    {
+        UpdateAssemblyInfoFilePath = "./src/LitJson/AssemblyInfo.cs",
+        UpdateAssemblyInfo = true,
+        OutputType = GitVersionOutput.Json,
+    });
+
+    version = assertedVersions.MajorMinorPatch;
+    semVersion = assertedVersions.LegacySemVerPadded;
+    milestone = string.Concat("v", version);
+
+    Information("Calculated Semantic Version: {0}", semVersion);
+
+    msBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("Version", semVersion)
+                            .WithProperty("AssemblyVersion", version)
+                            .WithProperty("FileVersion", version);
+
+    if(!IsRunningOnWindows())
+    {
+        var frameworkPathOverride = new FilePath(typeof(object).Assembly.Location).GetDirectory().FullPath + "/";
+
+        // Use FrameworkPathOverride when not running on Windows.
+        Information("Build will use FrameworkPathOverride={0} since not building on Windows.", frameworkPathOverride);
+        msBuildSettings.WithProperty("FrameworkPathOverride", frameworkPathOverride);
+    }
+
+    // Executed BEFORE the first task.
+    Information("Running tasks...");
+});
+
+Teardown(ctx =>
+{
+    // Executed AFTER the last task.
+    Information("Finished running tasks.");
+});
+
+///////////////////////////////////////////////////////////////////////////////
+// TASKS
+///////////////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Does(() => {
+    CleanDirectories(
+        new[] {
+            "./src/bin",
+            "./test/bin",
+            "./src/obj",
+            "./test/obj",
+            "./artifacts/nuget"   
+        }
+    );
+});
+
+Task("Restore")
+    .IsDependentOn("Clean")
+    .Does(() => {
+    DotNetCoreRestore("./LitJSON.sln",
+        new DotNetCoreRestoreSettings { MSBuildSettings = msBuildSettings }
+    );
+});
+
+Task("Build")
+    .IsDependentOn("Restore")
+    .Does(() => {
+    DotNetCoreBuild("./LitJSON.sln",
+        new DotNetCoreBuildSettings {
+            Configuration = configuration,
+            MSBuildSettings = msBuildSettings
+        }
+    );
+});
+
+Task("Test")
+    .IsDependentOn("Build")
+    .Does(() => {
+    DotNetCoreTest("./test/LitJSON.Tests.csproj",
+        new DotNetCoreTestSettings {
+            Configuration = configuration,
+            Framework = "netcoreapp2.0",
+            NoBuild = true
+        }
+    );
+
+    NUnit3("./test/**/bin/" + configuration + "/net45/*.Tests.dll", new NUnit3Settings {
+        NoResults = true
+    });
+});
+
+Task("Package")
+    .IsDependentOn("Test")
+    .Does(() => {
+    DotNetCorePack("./src/LitJSON/LitJSON.csproj",
+        new DotNetCorePackSettings {
+            Configuration = configuration,
+            NoBuild = true,
+            IncludeSymbols = true,
+            OutputDirectory = "./artifacts/nuget",
+            MSBuildSettings = msBuildSettings
+        }
+    );
+});
+
+Task("Upload-AppVeyor-Artifacts")
+    .IsDependentOn("Package")
+    .WithCriteria(AppVeyor.IsRunningOnAppVeyor)
+    .Does(() => {
+
+    foreach(var artifact in GetFiles("./artifacts/**/*.*"))
+    {
+        AppVeyor.UploadArtifact(artifact);
+    }
+});
+
+Task("Publish-MyGet")
+    .IsDependentOn("Package")
+    .WithCriteria((AppVeyor.IsRunningOnAppVeyor && !AppVeyor.Environment.PullRequest.IsPullRequest)
+        || StringComparer.OrdinalIgnoreCase.Equals(target, "Publish-MyGet"))
+    .Does(() => {
+
+      // Resolve the API key.
+    var apiKey = EnvironmentVariable("MYGET_API_KEY");
+    if(string.IsNullOrEmpty(apiKey)) {
+        throw new InvalidOperationException("Could not resolve MyGet API key.");
+    }
+
+    // Resolve the API url.
+    var apiUrl = EnvironmentVariable("MYGET_API_URL");
+    if(string.IsNullOrEmpty(apiUrl)) {
+        throw new InvalidOperationException("Could not resolve MyGet API url.");
+    }
+    
+    foreach(var package in (GetFiles("./artifacts/nuget/*.nupkg") - GetFiles("./artifacts/nuget/*.symbols.nupkg")))
+    {
+        DotNetCoreNuGetPush(package.FullPath,
+        new DotNetCoreNuGetPushSettings {
+            ApiKey = apiKey,
+            Source = apiUrl
+        }
+    );
+    }
+});
+
+Task("Publish-NuGet")
+    .IsDependentOn("Package")
+    .WithCriteria((AppVeyor.IsRunningOnAppVeyor && AppVeyor.Environment.Repository.Tag.IsTag && !AppVeyor.Environment.PullRequest.IsPullRequest)
+        || StringComparer.OrdinalIgnoreCase.Equals(target, "Publish-NuGet"))
+    .Does(() => {
+
+      // Resolve the API key.
+    var apiKey = EnvironmentVariable("NUGET_API_KEY");
+    if(string.IsNullOrEmpty(apiKey)) {
+        throw new InvalidOperationException("Could not resolve MyGet API key.");
+    }
+
+    // Resolve the API url.
+    var apiUrl = EnvironmentVariable("NUGET_API_URL");
+    if(string.IsNullOrEmpty(apiUrl)) {
+        throw new InvalidOperationException("Could not resolve MyGet API url.");
+    }
+    
+    foreach(var package in (GetFiles("./artifacts/nuget/*.nupkg") - GetFiles("./artifacts/nuget/*.symbols.nupkg")))
+    {
+        DotNetCoreNuGetPush(package.FullPath,
+        new DotNetCoreNuGetPushSettings {
+            ApiKey = apiKey,
+            Source = apiUrl
+        }
+    );
+    }
+});
+
+Task("AppVeyor")
+  .IsDependentOn("Upload-AppVeyor-Artifacts")
+  .IsDependentOn("Publish-MyGet")
+  .IsDependentOn("Publish-NuGet");
+
+Task("Default")
+  .IsDependentOn("Package");
+
+RunTarget(target);

--- a/build.ps1
+++ b/build.ps1
@@ -1,0 +1,83 @@
+$CakeVersion = "0.24.0"
+$DotNetChannel = "LTS";
+$DotNetVersion = "2.1.3";
+$DotNetInstallerUri = "https://raw.githubusercontent.com/dotnet/cli/v2.1.3/scripts/obtain/dotnet-install.ps1";
+$ENV:CAKE_NUGET_USEINPROCESSCLIENT='true'
+
+# Make sure tools folder exists
+$PSScriptRoot = Split-Path $MyInvocation.MyCommand.Path -Parent
+$ToolPath = Join-Path $PSScriptRoot "tools"
+if (!(Test-Path $ToolPath)) {
+    Write-Verbose "Creating tools directory..."
+    New-Item -Path $ToolPath -Type directory | out-null
+}
+
+###########################################################################
+# INSTALL .NET CORE CLI
+###########################################################################
+
+Function Remove-PathVariable([string]$VariableToRemove)
+{
+    $path = [Environment]::GetEnvironmentVariable("PATH", "User")
+    if ($path -ne $null)
+    {
+        $newItems = $path.Split(';', [StringSplitOptions]::RemoveEmptyEntries) | Where-Object { "$($_)" -inotlike $VariableToRemove }
+        [Environment]::SetEnvironmentVariable("PATH", [System.String]::Join(';', $newItems), "User")
+    }
+
+    $path = [Environment]::GetEnvironmentVariable("PATH", "Process")
+    if ($path -ne $null)
+    {
+        $newItems = $path.Split(';', [StringSplitOptions]::RemoveEmptyEntries) | Where-Object { "$($_)" -inotlike $VariableToRemove }
+        [Environment]::SetEnvironmentVariable("PATH", [System.String]::Join(';', $newItems), "Process")
+    }
+}
+
+# Get .NET Core CLI path if installed.
+$FoundDotNetCliVersion = $null;
+if (Get-Command dotnet -ErrorAction SilentlyContinue) {
+    $FoundDotNetCliVersion = dotnet --version;
+}
+
+if($FoundDotNetCliVersion -ne $DotNetVersion) {
+    $InstallPath = Join-Path $PSScriptRoot ".dotnet"
+    if (!(Test-Path $InstallPath)) {
+        mkdir -Force $InstallPath | Out-Null;
+    }
+    (New-Object System.Net.WebClient).DownloadFile($DotNetInstallerUri, "$InstallPath\dotnet-install.ps1");
+    & $InstallPath\dotnet-install.ps1 -Channel $DotNetChannel -Version $DotNetVersion -InstallDir $InstallPath;
+
+    Remove-PathVariable "$InstallPath"
+    $env:PATH = "$InstallPath;$env:PATH"
+    $env:DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+    $env:DOTNET_CLI_TELEMETRY_OPTOUT=1
+}
+
+###########################################################################
+# INSTALL CAKE
+###########################################################################
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+Function Unzip
+{
+    param([string]$zipfile, [string]$outpath)
+
+    [System.IO.Compression.ZipFile]::ExtractToDirectory($zipfile, $outpath)
+}
+
+
+# Make sure Cake has been installed.
+$CakePath = Join-Path $ToolPath "Cake.$CakeVersion/Cake.exe"
+if (!(Test-Path $CakePath)) {
+    Write-Host "Installing Cake $CakeVersion..."
+     (New-Object System.Net.WebClient).DownloadFile("https://www.nuget.org/api/v2/package/Cake/$CakeVersion", "$ToolPath\Cake.zip")
+     Unzip "$ToolPath\Cake.zip" "$ToolPath/Cake.$CakeVersion"
+     Remove-Item "$ToolPath\Cake.zip"
+}
+
+###########################################################################
+# RUN BUILD SCRIPT
+###########################################################################
+
+& "$CakePath" $args
+exit $LASTEXITCODE

--- a/build.sh
+++ b/build.sh
@@ -1,0 +1,60 @@
+#!/usr/bin/env bash
+##########################################################################
+# This is the Cake bootstrapper script for Linux and OS X.
+# This file was downloaded from https://github.com/cake-build/resources
+# Feel free to change this file to fit your needs.
+##########################################################################
+
+# Define directories.
+SCRIPT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+TOOLS_DIR=$SCRIPT_DIR/tools
+CAKE_VERSION=0.24.0
+CAKE_EXE=$TOOLS_DIR/Cake.$CAKE_VERSION/Cake.exe
+export CAKE_NUGET_USEINPROCESSCLIENT="true"
+
+# Make sure the tools folder exist.
+if [ ! -d "$TOOLS_DIR" ]; then
+  mkdir "$TOOLS_DIR"
+fi
+
+###########################################################################
+# INSTALL .NET CORE CLI
+###########################################################################
+
+echo "Installing .NET CLI..."
+if [ ! -d "$SCRIPT_DIR/.dotnet" ]; then
+  mkdir "$SCRIPT_DIR/.dotnet"
+fi
+curl -Lsfo "$SCRIPT_DIR/.dotnet/dotnet-install.sh" https://raw.githubusercontent.com/dotnet/cli/v2.1.3/scripts/obtain/dotnet-install.sh
+sudo bash "$SCRIPT_DIR/.dotnet/dotnet-install.sh" --version 2.1.3 --install-dir .dotnet --no-path
+export PATH="$SCRIPT_DIR/.dotnet":$PATH
+export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
+export DOTNET_CLI_TELEMETRY_OPTOUT=1
+"$SCRIPT_DIR/.dotnet/dotnet" --info
+
+
+###########################################################################
+# INSTALL CAKE
+###########################################################################
+
+if [ ! -f "$CAKE_EXE" ]; then
+    echo "Installing Cake $CAKE_VERSION..."
+    curl -Lsfo Cake.zip "https://www.nuget.org/api/v2/package/Cake/$CAKE_VERSION" && unzip -q Cake.zip -d "$TOOLS_DIR/Cake.$CAKE_VERSION" && rm -f Cake.zip
+    if [ $? -ne 0 ]; then
+        echo "An error occured while installing Cake."
+        exit 1
+    fi
+fi
+
+# Make sure that Cake has been installed.
+if [ ! -f "$CAKE_EXE" ]; then
+    echo "Could not find Cake.exe at '$CAKE_EXE'."
+    exit 1
+fi
+
+###########################################################################
+# RUN BUILD SCRIPT
+###########################################################################
+
+# Start Cake
+exec mono "$CAKE_EXE" "$@"

--- a/src/LitJson/JsonException.cs
+++ b/src/LitJson/JsonException.cs
@@ -14,7 +14,12 @@ using System;
 
 namespace LitJson
 {
-    public class JsonException : ApplicationException
+    public class JsonException :
+#if NETSTANDARD1_5
+        Exception
+#else
+        ApplicationException
+#endif
     {
         public JsonException () : base ()
         {

--- a/src/LitJson/JsonMapper.cs
+++ b/src/LitJson/JsonMapper.cs
@@ -314,9 +314,15 @@ namespace LitJson
             Type value_type = underlying_type ?? inst_type;
 
             if (reader.Token == JsonToken.Null) {
+                #if NETSTANDARD1_5
+                if (inst_type.IsClass() || underlying_type != null) {
+                    return null;
+                }
+                #else
                 if (inst_type.IsClass || underlying_type != null) {
                     return null;
                 }
+                #endif
 
                 throw new JsonException (String.Format (
                             "Can't assign null to an instance of type {0}",
@@ -357,9 +363,13 @@ namespace LitJson
                 }
 
                 // Maybe it's an enum
+                #if NETSTANDARD1_5
+                if (value_type.IsEnum())
+                    return Enum.ToObject (value_type, reader.Value);
+                #else
                 if (value_type.IsEnum)
                     return Enum.ToObject (value_type, reader.Value);
-
+                #endif
                 // Try using an implicit conversion operator
                 MethodInfo conv_op = GetConvOp (value_type, json_type);
 

--- a/src/LitJson/JsonReader.cs
+++ b/src/LitJson/JsonReader.cs
@@ -395,7 +395,9 @@ namespace LitJson
             end_of_json  = true;
 
             if (reader_is_owned)
-                reader.Close ();
+            {
+                using(reader){}
+            }
 
             reader = null;
         }

--- a/src/LitJson/LitJSON.csproj
+++ b/src/LitJson/LitJSON.csproj
@@ -1,0 +1,30 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net45;netstandard1.5</TargetFrameworks>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <GenerateAssemblyInfo>false</GenerateAssemblyInfo>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PackageId>LitJson</PackageId>
+    <Copyright>The authors disclaim copyright to this source code.</Copyright>
+    <Authors>Leonardo Boshell, Mattias Karlsson and contributors</Authors>
+    <Company>Leonardo Boshell, Mattias Karlsson and contributors</Company>
+    <PackageLicenseUrl>https://github.com/LitJSON/litjson/blob/develop/COPYING</PackageLicenseUrl>
+    <PackageIconUrl>https://litjson.net/assets/img/logo.png</PackageIconUrl>
+    <RepositoryUrl>https://github.com/LitJSON/litjson</RepositoryUrl>
+    <RepositoryType>git</RepositoryType>
+    <PackageTags>JSON;Serializer</PackageTags>
+  </PropertyGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.5' ">
+    <PackageReference Include="System.Collections.NonGeneric" Version="4.0.1" />
+    <PackageReference Include="System.Collections.Specialized" Version="4.0.1" />
+    <PackageReference Include="System.Reflection" Version="4.3.0" />
+    <PackageReference Include="System.Reflection.TypeExtensions" Version="4.1.0" />
+  </ItemGroup>
+
+</Project>

--- a/src/LitJson/Netstandard15Polyfill.cs
+++ b/src/LitJson/Netstandard15Polyfill.cs
@@ -1,0 +1,24 @@
+#if NETSTANDARD1_5
+using System;
+using System.Reflection;
+namespace LitJson
+{
+    internal static class Netstandard15Polyfill
+    {
+        internal static Type GetInterface(this Type type, string name)
+        {
+            return type.GetTypeInfo().GetInterface(name); 
+        }
+
+        internal static bool IsClass(this Type type)
+        {
+            return type.GetTypeInfo().IsClass;
+        }
+
+        internal static bool IsEnum(this Type type)
+        {
+            return type.GetTypeInfo().IsEnum;
+        }
+    }
+}
+#endif

--- a/test/JsonDataTest.cs
+++ b/test/JsonDataTest.cs
@@ -194,27 +194,29 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (InvalidOperationException))]
         public void GetKeysInvalidTypeTest ()
         {
             JsonData data = new JsonData ();
             data.Add (42);  // turns it into an array
 
             // .. but an array doesn't have keys
-            ICollection<string> keys = data.Keys;
-            Assert.IsNotNull (keys);
+            Assert.Throws<InvalidOperationException>(() => {
+                ICollection<string> keys = data.Keys;
+                Assert.IsNotNull (keys);
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (InvalidCastException))]
         public void InvalidCastTest ()
         {
             JsonData data = 35;
 
-            string str = (string) data;
+            Assert.Throws<InvalidCastException>(() => {
+                string str = (string) data;
 
-            if (str != (string) data)
-                str = (string) data;
+                if (str != (string) data)
+                    str = (string) data;
+            });
         }
 
         [Test]

--- a/test/JsonMapperTest.cs
+++ b/test/JsonMapperTest.cs
@@ -465,7 +465,7 @@ namespace LitJson.Test
             Assembly asmb = typeof (JsonMapperTest).Assembly;
 
             StreamReader stream = new StreamReader (
-                asmb.GetManifestResourceStream ("json-example.txt"));
+                asmb.GetManifestResourceStream (asmb.GetName().Name + ".json-example.txt"));
 
             using (stream) {
                 data = JsonMapper.ToObject (stream);
@@ -753,7 +753,6 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ImportObjectNonMembersTest()
         {
             string json = @"
@@ -771,12 +770,13 @@ namespace LitJson.Test
             JsonReader reader = new JsonReader(json);
             reader.SkipNonMembers = false;
 
-            UiWindow window = JsonMapper.ToObject<UiWindow>(reader);
-            window.title = "Unreachable";
+            Assert.Throws<JsonException>(() => {
+                UiWindow window = JsonMapper.ToObject<UiWindow>(reader);
+                window.title = "Unreachable";
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ImportStrictCommentsTest ()
         {
             string json = @"
@@ -790,14 +790,15 @@ namespace LitJson.Test
             JsonReader reader = new JsonReader (json);
             reader.AllowComments = false;
 
-            JsonData data = JsonMapper.ToObject (reader);
+            Assert.Throws<JsonException>(() => {
+                JsonData data = JsonMapper.ToObject (reader);
 
-            if (data.Count != 3)
-                data = JsonMapper.ToObject (reader);
+                if (data.Count != 3)
+                    data = JsonMapper.ToObject (reader);
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ImportStrictStringsTest ()
         {
             string json = "[ 'Look! Single quotes' ]";
@@ -805,10 +806,12 @@ namespace LitJson.Test
             JsonReader reader = new JsonReader (json);
             reader.AllowSingleQuotedStrings = false;
 
-            JsonData data = JsonMapper.ToObject (reader);
+            Assert.Throws<JsonException>(() => {
+                JsonData data = JsonMapper.ToObject (reader);
 
-            if (data[0] == null)
-                data = JsonMapper.ToObject (reader);
+                if (data[0] == null)
+                    data = JsonMapper.ToObject (reader);
+            });
         }
 
         [Test]

--- a/test/JsonReaderTest.cs
+++ b/test/JsonReaderTest.cs
@@ -161,7 +161,7 @@ namespace LitJson.Test
         {
             Assembly asmb = typeof (JsonReaderTest).Assembly;
             StreamReader stream = new StreamReader (
-                asmb.GetManifestResourceStream ("json-example.txt"));
+                asmb.GetManifestResourceStream (asmb.GetName().Name + ".json-example.txt"));
 
             JsonReader reader = new JsonReader (stream);
 
@@ -191,18 +191,18 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void LexerErrorEscapeSequenceTest ()
         {
             string json = "[ \"Hello World \\ufffg \" ]";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void LexerErrorRealNumberTest ()
         {
             // One ore more digits have to appear after the '.'
@@ -210,18 +210,21 @@ namespace LitJson.Test
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void LexerErrorTrueTest ()
         {
             string json = "[ TRUE ]";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
@@ -276,13 +279,13 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (ArgumentNullException))]
         public void NullReaderTest ()
         {
             TextReader text_reader = null;
-            JsonReader reader = new JsonReader (text_reader);
 
-            while (reader.Read ());
+            Assert.Throws<ArgumentNullException>(() => {
+                JsonReader reader = new JsonReader (text_reader);
+            });
         }
 
         [Test]
@@ -300,40 +303,42 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ParserErrorArrayClosingTest ()
         {
             string json = "[ 1, 2, 3 }";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ParserErrorIncompleteObjectTest ()
         {
             string json = "{ \"temperature\" : 21 ";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ParserErrorNoArrayOrObjectTest ()
         {
             string json = "true";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ParserErrorObjectClosingTest ()
         {
             string json = @"{
@@ -342,18 +347,21 @@ namespace LitJson.Test
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ParserErrorPropertyExpectedTest ()
         {
             string json = "{ {\"foo\": bar} }";
 
             JsonReader reader = new JsonReader (json);
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
@@ -414,7 +422,6 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void StrictCommentsTest ()
         {
             string json = @"
@@ -428,11 +435,12 @@ namespace LitJson.Test
             JsonReader reader = new JsonReader (json);
             reader.AllowComments = false;
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void StrictStringsTest ()
         {
             string json = "[ 'Look! Single quotes' ]";
@@ -440,7 +448,9 @@ namespace LitJson.Test
             JsonReader reader = new JsonReader (json);
             reader.AllowSingleQuotedStrings = false;
 
-            while (reader.Read ());
+            Assert.Throws<JsonException>(() => {
+                while (reader.Read ());
+            });
         }
 
         [Test]

--- a/test/JsonWriterTest.cs
+++ b/test/JsonWriterTest.cs
@@ -35,7 +35,6 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorExcessDataTest ()
         {
             JsonWriter writer = new JsonWriter ();
@@ -43,31 +42,34 @@ namespace LitJson.Test
             writer.WriteArrayStart ();
             writer.Write (true);
             writer.WriteArrayEnd ();
-            writer.Write (false);
+            Assert.Throws<JsonException>(() => {
+                writer.Write (false);
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorArrayClosingTest ()
         {
             JsonWriter writer = new JsonWriter ();
 
             writer.WriteArrayStart ();
             writer.Write (true);
-            writer.WriteObjectEnd ();
+            Assert.Throws<JsonException>(() => {
+                writer.WriteObjectEnd ();
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorNoArrayOrObjectTest ()
         {
             JsonWriter writer = new JsonWriter ();
 
-            writer.Write (true);
+            Assert.Throws<JsonException>(() => {
+                writer.Write (true);
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorObjectClosingTest ()
         {
             JsonWriter writer = new JsonWriter ();
@@ -75,29 +77,32 @@ namespace LitJson.Test
             writer.WriteObjectStart ();
             writer.WritePropertyName ("foo");
             writer.Write ("bar");
-            writer.WriteArrayEnd ();
+            Assert.Throws<JsonException>(() => {
+                writer.WriteArrayEnd ();
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorPropertyExpectedTest ()
         {
             JsonWriter writer = new JsonWriter ();
 
             writer.WriteObjectStart ();
-            writer.Write (10);
-            writer.WriteObjectEnd ();
+            Assert.Throws<JsonException>(() => {
+                writer.Write (10);
+            });
         }
 
         [Test]
-        [ExpectedException (typeof (JsonException))]
         public void ErrorValueExpectedTest ()
         {
             JsonWriter writer = new JsonWriter ();
 
             writer.WriteObjectStart ();
             writer.WritePropertyName ("foo");
-            writer.WriteObjectEnd ();
+            Assert.Throws<JsonException>(() => {
+                writer.WriteObjectEnd ();
+            });
         }
 
         [Test]
@@ -153,13 +158,13 @@ namespace LitJson.Test
         }
 
         [Test]
-        [ExpectedException (typeof (ArgumentNullException))]
         public void NullWriterTest ()
         {
             TextWriter text_writer = null;
-            JsonWriter writer = new JsonWriter (text_writer);
 
-            writer.Write (123);
+            Assert.Throws<ArgumentNullException>(() => {
+                JsonWriter writer = new JsonWriter (text_writer);
+            });
         }
 
         [Test]

--- a/test/LitJSON.Tests.csproj
+++ b/test/LitJSON.Tests.csproj
@@ -1,0 +1,23 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp2.0;net45</TargetFrameworks>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="nunit" Version="3.9.0" />
+    <PackageReference Include="NUnit3TestAdapter" Version="3.9.0" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.5.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\src\LitJson\LitJSON.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="**/*.txt" />
+  </ItemGroup>
+
+</Project>

